docs/spec.md

```diff
@@ -538,6 +561,19 @@ All integer types explicitly specified (uint32, uint64, etc.) in this section ar
 | _encoding length_ | char[]  | encoding, same encoding that was used for the request |
 | remaining bytes   | uint8[] | response payload                                      |
 
+### Asset
+
+- Response to a [fetch asset](#fetch-asset) request
+- Only supported if the server previously declared the `assets` [capability](#server-info).
+
+| Bytes           | Type             | Description                                                                                                                                              |
+| --------------- | ---------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| 1               | opcode           | 0x04                                                                                                                                                     |
+| 4               | uint32           | request id, as given in the corresponding [request](#fetch-asset)                                                                                        |
+| 1               | uint8            | status enum, `0` for success and `1` for error. Values `>1` are reserved for future use                                                                  |
+| 4 + N           | uint32 + char[N] | asset [media type](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types), if `status == 0` <br /> error message, if `status == 1` |
```

Comment:
I find this a bit weird that this field is overloaded with the error message and the media type. Is there a problem with using a separate error field that is empty if there is no error? The length prefix overhead?

Another option would be two opcodes (one for assets success and another for error) not sure if this was discussed.

Reply:
> Is there a problem with using a separate error field that is empty if there is no error? The length prefix overhead?

Changed it to use a separate field.